apps/portal-api/src/thirdparty/elasticsearch/client.ts

```diff
@@ -0,0 +1,99 @@
+import { Client, ClientOptions } from '@elastic/elasticsearch';
+import IndicesApi from '@elastic/elasticsearch/lib/api/api/indices.js';
+import {
+  CreateRequest,
+  CreateResponse,
+  DeleteRequest,
+  DeleteResponse,
+  IndexRequest,
+  IndexResponse,
+  QueryDslQueryContainer,
+  SearchRequest,
+  SearchResponse,
+  Sort,
+} from '@elastic/elasticsearch/lib/api/types.js';
+import portalConfig from '../../config';
+import { logApp } from '../../utils/app-logger.util';
+
+export class ElasticSearchService {
+  private elasticsearchClient: Client;
+
+  constructor() {
+    const config: ClientOptions = {
+      node: `${portalConfig.elasticsearch.protocol}://${portalConfig.elasticsearch.host}:${portalConfig.elasticsearch.port}`,
+    };
+
+    config.auth = {
+      username: portalConfig.elasticsearch.username,
+      password: portalConfig.elasticsearch.password,
+    };
+
+    this.elasticsearchClient = new Client(config);
+  }
+
+  getIndices(): IndicesApi {
+    return this.elasticsearchClient.indices;
+  }
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  async search<T = any>(params: SearchRequest): Promise<SearchResponse<T>> {
```

Comment:
Does it work with eslint if T is `unknown` here ? (To remove the eslint-disable line above 😉)

Or maybe `Record<string, unknown>` ?